### ISSUE TITLE
RNN layer does not reset dropout masks of RNNCell

### ISSUE THREAD

**Issue by ExpectationMax** [tensorflow/tensorflow#32543]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): macOS Mojave 10.14.6
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below):  v1.12.1-9392-gf3c7314d83 1.15.0-rc0
- Python version: 3.7.4

**Describe the current behavior**
The RNN layer with an RNNCell does not reset the states of dropout masks compared to the layer implementations of the cells. Thus the behavior of `tf.keras.layers.GRU(10)` != `tf.keras.layers.RNN(tf.keras.layers.GRUCell(10))`.

This is especially problematic, because the [Keras RNN API tutorial](https://www.tensorflow.org/beta/guide/keras/rnn#rnn_layers_and_rnn_cells) states both approaches are mathematically equivalent.

**Describe the expected behavior**
The RNN layer should check the type of the RNNCell and, if it is a subclass of `DropoutRNNCellMixin`, reset the dropout masks after each call. By calling `cell.reset_recurrent_dropout_mask()` and `cell.reset_dropout_mask`.

**Code to reproduce the issue**
Partially copied from https://github.com/tensorflow/tensorflow/issues/29391
```python
from __future__ import absolute_import, division, print_function
import numpy as np

import tensorflow as tf
tf.enable_eager_execution()


tf.enable_eager_execution()
print(tf.__version__)
data = np.random.normal(0, 1, (1, 10, 2)).astype(np.float32)
rnn = tf.keras.layers.GRU(units=10, dropout=0.5,
                                   recurrent_dropout=0.5)
print(set([rnn(data, training=True).numpy()[0, 0] for _ in range(5)]))

rnn_cell = tf.keras.layers.GRUCell(units=10, dropout=0.5,
                                   recurrent_dropout=0.5)
rnn = tf.keras.layers.RNN(rnn_cell)
print(set([rnn(data, training=True).numpy()[0, 0] for _ in range(5)]))
```

Output:
```
WARNING:tensorflow:From check_dropout.py:5: The name tf.enable_eager_execution is deprecated. Please use tf.compat.v1.enable_eager_execution instead.

1.15.0-rc0
{0.04537238, 0.15487108, 0.0, 0.08881481, 0.055508718}  # Different dropout mask was used for each call
{-0.34464198}  # Same dropout mask was used for each call
```

**Comment by ravikyram** [comment]
I have tried on colab with TF 1.15.0-rc0 , nightly versions and was able to reproduce the issue.Please, find the gist [here](https://colab.sandbox.google.com/gist/ravikyram/d29fc400d2933fbf3efedfe1311f9a83/untitled186.ipynb).Thanks!

**Comment by qlzh727** [comment]
Thanks for reporting the issue, let me fix it.
